sqsworkers/crew.py

```diff
@@ -156,35 +164,69 @@ def run(self):
         self.poll_queue()
 
     def poll_queue(self):
```

Comment:
Rather than re-writing the implementation of this class, might be worth looking into subclassing and overwriting the methods in your subclass or having private methods that you dispatch to from the public interface based on `bulk_mode`.

Reply:
I will come up with a subclassed version later today

Reply:
I looked at this. Presently there isn't enough time to implement this, but I will try to keep a version ready for the next time I touch it.

Reply:
If you can't figure out how to subclass and override the `poll_queue` method, rename the original `poll_queue` to `_poll_queue`, write your logic in a `_poll_queue_bulk` method, and have the "public" `poll_queue` method have something along the lines of:
```python
if self.bulk_mode:
  self._poll_queue_bulk()
else:
  self._poll_queue()
```

Reply:
Thanks, I will try this. It's not just about subclassing and overriding. There were some runtime errors owing perhaps to the way members were accessed, or some members not being available to the child class at the time they were accessed. But the way you said here is will not have that problem